docker-compose.prod.yml

```diff
@@ -0,0 +1,99 @@
+version: "3.7"
+
+x-web: &web
+  image: trojsten/web:latest
+  build:
+    context: .
+    dockerfile: ./docker/prod/Dockerfile
+  env_file:
+    - ./docker/prod/web.env
+    - production.env
+
+services:
+  db:
+    image: postgres:12-alpine
+    volumes:
+      - db_data:/var/lib/postgresql/data
+    environment:
+      - POSTGRES_USER=trojsten
+      - POSTGRES_DB=trojsten
+  static:
+    <<: *web
+    ports:
+      - "8000:80"
+    command: "nginx -g 'pid /tmp/nginx.pid; daemon off;'"
+  login:
+    <<: *web
+    ports:
+      - "8010:80"
+    depends_on:
+      - db
+    environment:
+      - DJANGO_SETTINGS_MODULE=trojsten.settings.login
+      - TROJSTENWEB_ALLOWED_HOSTS=login.trojsten.sk;betalogin.trojsten.sk;login;localhost;127.0.0.1;
+  ksp:
+    <<: *web
+    ports:
+      - "8001:80"
+    environment:
+      - DJANGO_SETTINGS_MODULE=trojsten.settings.ksp
+      - TROJSTENWEB_ALLOWED_HOSTS=ksp.sk;beta.ksp.sk;localhost;127.0.0.1;
```

Comment:
is this file meant as a real production configuration, or just as an example? if real production:
- shouldn't we have separate configuration for beta? 
- KSP website is currently based on www.ksp.sk
- production database is currently hosted on inteligent.. are we changing this to local postgresql instance inside a docker-compose group?

Reply:
> is this file meant as a real production configuration, or just as an example? if real production:

This was meant as actual production config. Maybe first to be used just with beta.

> * shouldn't we have separate configuration for beta?

Maybe, maybe not. For domains there is no harm in including both in the whitelist.
If we use same config for both, it's easier to manage. We can read the different settings in .env file.

> * KSP website is currently based on [www.ksp.sk](http://www.ksp.sk)

Maybe we could change it to ksp.sk to be consistent? Or should we change others to have www? If we want to keep ksp with www and others without, we can change this config.

> * production database is currently hosted on inteligent.. are we changing this to local postgresql instance inside a docker-compose group?

I think we should, so we can host the whole thing from cloud. It could be a bit faster if hosted in the same cluster/machine.

But if we have a reason to use inteligent hosted db, lmk, I can adapt the config.

Reply:
Please, don't assume anything about how exactly this web will be hosted, on which domain or which DB it will use (at least in this repo). Our config management code should be a single source of truth for that. This particular docker compose file will be useful as an example, but the real one will be in our config management repo, so that we may update it easily as the underlying infrastructure changes.

Reply:
When deploying in the cloud, I am used to having databases on a managed instance (Cloud SQL) in the same datacenter and deploying into a K8s cluster from a K8s configuration file. I didn't even know it was possible to deploy from a compose file. Let's leave it like this for now and we can potentially rewrite this during the actual deployment.

Reply:
@black3r leaving this as-is could be confusing, as this won't be used in prod, and its name suggests so. (The real config, either K8s config together with whatever is required to configure a CloudSQL instance, or file similar to this for use in docker swarm, will be in our config management repo.) This file is however a good starting point so I'd suggest renaming it to `docker-compose.prod.example.yml` or sth.

Reply:
if the configuration files will be in the central config management repo, how will we handle continuous deployment?

Reply:
Sounds good. I didn't realize we store the prod config in separate repo. I will rename this to example and simplify a bit. The main purpose of this file will be to verify the prod docker file locally.

Reply:
> how will we handle continuous deployment?

I don't have much experience with k8s, so correct me if I'm wrong, but I imagined we will have `imagePullPolicy: Always` in the config and create deployment with config from our config management repo. From CI of this repo we will just trigger [`kubectl rollout restart`](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#-em-restart-em-) which should pull a new image in that case.